travelwise/frontend/views.py

```diff
@@ -114,4 +111,29 @@ def history_view(request):
         
         return render(request, 'travel/history.html', {
             'travel_plans': travel_plans
-        })
\ No newline at end of file
+
+
+@login_required(login_url="landing")
+def profile_view(request):
+    if request.method == 'POST':
+        request.user.first_name = request.POST.get('first_name')
+        request.user.last_name = request.POST.get('last_name')
+        request.user.save()
+
+        if not request.POST.get('new_password') and not request.POST.get('new_password'):
+            return render(request, 'accounts/profile.html', {'form': {}})
+
+        data = {
+            'new_password1': request.POST.get('new_password'),
+            'new_password2': request.POST.get('confirm_password')
+        }
+
+        form = SetPasswordForm(user=request.user, data=data)
+        if form.is_valid():
+            form.save()
+    else:
+        form = {}
```

Comment:
Probably should be a 405 error.

Reply:
I want it to also work for a get request, when they are just viewing the page.

Reply:
Updated. Let me know if this looks better.